ip/trie.go

```diff
@@ -93,7 +93,7 @@ func (t *V4Trie) Get(cidr V4CIDR) interface{} {
 }
 
 func (t *V4Trie) LookupPath(buffer []V4TrieEntry, cidr V4CIDR) []V4TrieEntry {
-	return t.root.lookupPath(buffer, cidr)
+	return t.root.lookupPath(buffer[:0], cidr)
```

Comment:
I wondered why `buffer` was passed into `LookupPath`.  What is happening here?  Is `buffer[:0]` equivalent to `[]V4TrieEntry{}`, and hence `buffer` isn't needed any more?

Reply:
It's an optimisation, `[]buffer[:0]` makes a new slice that uses the same backing array as buffer but has zero length, that means that appending to the buffer will avoid allocations as long as the backing array has sufficient capacity.  It's a pattern that gets used in the standard library, usually passing in a byte buffer for the library function to put its data in.  If you pass nil, it'll append to nil, which creates a new slice.

Reply:
Interesting, thanks for explaining!